frame/contracts/src/exec.rs

```diff
@@ -872,6 +909,13 @@ where
 				}
 			}
 		} else {
+			if let Some(message) = &self.debug_message {
```

Comment:
Correct me if I'm wrong, but wouldn't this result in the same messages being repeated with nested contract calls?

Reply:
No: It is inside a branch that is only taken when the `first_frame` is popped. i.e when returning from the "root call".